src/java.base/share/classes/java/util/Collections.java

```diff
@@ -5798,4 +5798,61 @@ public boolean removeIf(Predicate<? super E> filter) {
         @Override
         public Stream<E> parallelStream()   {return q.parallelStream();}
     }
+
+    /**
+     * Calculate initial capacity for HashMap based classes, from expected size.
+     *
+     * @param expectedSize expected size
+     * @return initial capacity for HashMap based classes.
+     * @since 19
+     */
+    private static int calculateHashMapCapacity(int expectedSize) {
+        if (expectedSize >= 1610612736) {
+            return Integer.MAX_VALUE;
+        }
+        return (expectedSize + (expectedSize + 2) / 3);
+    }
+
+    /**
+     * Create new HashMap from expected size.
+     *
+     * @param expectedSize expected size
+     * @param <K>          the type of keys maintained by this map
+     * @param <V>          the type of mapped values
+     * @return initial capacity for HashMap based classes.
```

Comment:
That isn't what is returned.

Reply:
> That isn't what is returned.

Yes it isn't actually. but I cannot find a better description string for it.

"the pre-processed raw initial capacity for HashMap based classes." is more exact but sounds weird.

Have you any suggestions to replace this? Thanks!

Reply:
This method doesn't return the capacity it returns the Map!

Reply:
my mistake. will fix it in hours.

Reply:
@dholmes-ora done.